### PR TITLE
Fix to show payment info in completed campaigns

### PR DESCRIPTION
**Pull request by therocket-gr**
Related to #
SELFSERVE-574

## Proposed Changes

Not sure if we have a campaign status of "completed" in the result i see the "finished" status.. 

i added a || (or) condition to display payment info 

## Testing Instructions

checkout branch and check a completed campaign 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx [hunk 1]
```diff
@@ -504,8 +504,7 @@ export default function CampaignItemDetails( props: Props ) {
 								</div>
 							</div>
 						</div>
-
-						{ status === 'completed' && (
+						{ ( status === 'completed' || status === 'finished' ) && (
```
Comment (jjolmo) [inline client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx]: I'm guessing the 'completed' status doesn't exist. It should be just 'finished`. but I guess other status might want to see this information.

Campaign canceled: User might have been billed when they cancel the campaign if some impressions got delivered

so I'm guessing it should be:

```
status === 'finished' || status === 'canceled'
```

Comment (therocket-gr) [reply client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx]: fixed.. please check and test it again...

**Review by timur987 — APPROVED** [review]
Thanks @therocket-gr!

The changes LGTM and work as advertised:
image

I used the following data to override reading `billing_data` ([here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx#L100)):
```js
const { card_name, payment_method, subtotal, credits, total } = {
	card_name: 'Timur Gogolev',
	payment_method: 'Visa **** 4242',
	subtotal: 75,
	credits: 25.25,
	total: 49.75,
};
```

I found the difference between the [design](https://www.figma.com/file/uvh2KtY65b7hGBHI9exbJ1/BlazePress?type=design&node-id=7223-61220&t=p02iL5QIGG1MjHED-0) and how `formatCents` function works:

In designs it shows cents part even if it's 0 like `$75,00` and there's `,` (comma) as a separator.
Screenshot 2023-06-16 at 15 19 18

In our case we have `$75` (no cents part provided), `$25.25` (like on the image below, if cents part is provided) and we use a dot as a separator:
Screenshot 2023-06-16 at 15 27 21

I'm mentioning it to show discrepancy between designs and our implementation but I totally fine with what we got.

**Review by sbarbosa — APPROVED** [review]
The code looks good and tested correctly 👍

**Pull request merged**